InterGalacticCurrencyConvertor/src/main/java/numerals/RomanNumerals.java

```diff
@@ -0,0 +1,174 @@
+                                          HashMap<Character, Integer> charCntMap) {
+    boolean passed = false;
+    if (String.valueOf(curChar).matches(MATCH_CHAR_VLD)) {
+      return passed;
+    } else {
+      int curVal = mRnToDecimalMap.get(curChar);
+      List<Character> repChars = Arrays.asList('i', 'x', 'c', 'm');
+      for(Character c : repChars) {
+        if (charCntMap.get(c) == 3) {
+          return passed;
+        }
+      }
+      for (int j = idx - 2; j >= 0; --j) {
+        if (values.get(j) < curVal) {
+          return passed;
+        }
+      }
+    }
+    return passed = true;
+  }
+
+	private boolean isValidSymbolList(List<String> rSymbolList)
+	                                  throws IgnConversionException {
+	  boolean isValid = false;
+    Objects.requireNonNull(rSymbolList, "Null Roman Numeral List Provided");
+    if (rSymbolList.size() == 0) {
+      throw new IllegalArgumentException("Empty Roman Numeral List");
+    }
+	  ArrayList<Integer> values = new ArrayList<>();
+	  int minimum = Integer.MAX_VALUE;
+	  int prevVal = minimum;
+	  Character prevChar = ' ';
+	  HashMap<Character, Integer> charCntMap = new HashMap<>();
+	  for(Character rSymbol : mRnToDecimalMap.keySet()) {
+	    charCntMap.put(rSymbol, 0);
+	  }
+	  for (int i = 0; i < rSymbolList.size(); ++i) {
+      if (!isValidSymbol(rSymbolList.get(i))) {
+        // This should never occur
+        throw new IllegalArgumentException("Invalid conversion symbol " + rSymbolList.get(i) +
+                                           " in list");
+      }
+	    Character curChar = rSymbolList.get(i).toLowerCase().charAt(0);
+	    int curVal = mRnToDecimalMap.get(curChar);
+	    if (prevVal < curVal) {
+	      if (!hasSubtractionRulePassed(curChar, prevChar, i, values)) {
+	        return isValid;
+	      }
+	    } else if (prevVal > curVal){
+	      if (!hasAdditionRulePassed(curChar, i, values)) {
+	        return isValid;
+	      }
+	    } else {
+	      if (!hasRepetitionRulePassed(curChar, i, values, charCntMap)) {
+	        return isValid;
+	      }
+	    }
+	    prevChar = curChar;
+	    prevVal = curVal;
+	    minimum = Math.min(minimum, curVal);
+	    values.add(curVal);
+	    int charOccurrence = charCntMap.get(curChar);
+	    charCntMap.put(curChar, charOccurrence + 1);
+	  }
+	  return isValid = true;
+	}
+
+	@Override
+  public boolean isValidSymbol(String rSymbol) throws IgnConversionException {
+	  Objects.requireNonNull(rSymbol, "Null Roman Numeral Provided");
+	  if(rSymbol.length() == 0) {
+	    throw new IllegalArgumentException("Invalid Roman Numeral " + rSymbol);
+	  } else if (rSymbol.length() > 1) {
+	    throw new IgnConversionException(IgnConversionErrors.INVALID_CONVERSION_SYMBOL_LENGTH,
+                                       "Invalid length of conversion symbol " +
+	                                     rSymbol);
+	  }
+		return mRnToDecimalMap.containsKey(rSymbol.toLowerCase().charAt(0));
+	}
+
+	@Override
```

Comment:
Just fix the formatting of the whole project. There is a way to do this in eclipse - search online.